pkg/kv/kvserver/store.go

```diff
@@ -1743,6 +1744,15 @@ func NewStore(
 	queueAdditionOnSystemConfigUpdateBurst.SetOnChange(&cfg.Settings.SV,
 		updateSystemConfigUpdateQueueLimits)
 
+	// Set up callback to update lock table size when the cluster setting changes.
+	concurrency.DefaultLockTableSize.SetOnChange(&cfg.Settings.SV, func(ctx context.Context) {
+		newSize := concurrency.DefaultLockTableSize.Get(&cfg.Settings.SV)
+		s.VisitReplicas(func(repl *Replica) bool {
```

Comment:
Nice. When I had tried to do something similar on a branch, I had each lock table call SetOnChange but that felt really bad. This seems a lot cleaner than that.